homeassistant/components/matter/entity.py

```diff
@@ -59,23 +59,18 @@ def __init__(
         self.entity_description = entity_description
         self._unsubscribes: list[Callable] = []
         # for fast lookups we create a mapping to the attribute paths
+        # The server info is set when the client connects to the server.
         self._attributes_map: dict[type, str] = {}
         server_info = cast(ServerInfo, self.matter_client.server_info)
-        # The server info is set when the client connects to the server.
-        assert server_info is not None
         # create unique_id based on "Operational Instance Name" and endpoint/device type
         self._attr_unique_id = (
             f"{get_operational_instance_id(server_info, self._node_device.node())}-"
             f"{device_type_instance.endpoint}-"
             f"{device_type_instance.device_type.device_type}"
         )
-
-    @property
-    def device_info(self) -> DeviceInfo | None:
-        """Return device info for device registry."""
-        server_info = cast(ServerInfo, self.matter_client.server_info)
-        dev_id = get_device_id(server_info, self._node_device)
-        return {"identifiers": {(DOMAIN, dev_id)}}
+        self._attr_device_info = {
```

Comment:
We normally use `DeviceInfo` explicitly to create the typed dict. Then we can use parameters instead of string keys to populate the dict.